src/Main/TodoCard.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useCallback, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/todo';
+import { titleChanger } from '../utils/functions';
+
+type Props = {
+  id: number;
+  completed: boolean;
+  title: string;
+  todosUpdater: (prevState: Todo[]) => void;
+  todos: Todo[];
+};
+
+export const TodoCard: React.FC<Props> = ({
+  id,
+  title,
+  completed,
+  todosUpdater,
+  todos,
+}) => {
+  const [todoOnEdit, setTodoOnEdit] = useState<Todo | null>(null);
+  const [titleQuery, setTitleQuery] = useState<string>('');
+
+  const callbackRef = useCallback((inputElement: HTMLInputElement) => {
+    if (inputElement) {
+      inputElement.focus();
+    }
+  }, []);
+
+  const handleCompletedChange = () => {
+    const todosCopy = [...todos].map(todo => (
+      (todo.id === id) ? { ...todo, completed: !completed } : todo
+    ));
+
+    return todosUpdater(todosCopy);
+  };
+
+  const handleTodoDelete = () => {
+    const todoDelete = todos.filter(todo => todo.id !== id);
+
+    return todosUpdater(todoDelete);
+  };
+
+  const handleDBClick = () => {
+    const todoToEdit = todos.find(todo => todo.id === id);
+
+    if (todoToEdit) {
+      setTodoOnEdit(todoToEdit);
+      setTitleQuery(todoToEdit.title);
+    }
+  };
+
+  const handleEnterPress = () => {
+    if (todoOnEdit) {
+      if (todoOnEdit.title === titleQuery) {
+        setTodoOnEdit(null);
+
+        return;
+      }
+
+      todosUpdater(titleChanger(todos, todoOnEdit, titleQuery));
+    }
+
+    setTodoOnEdit(null);
```

Comment:
You call setTodoOnEdit(null) in any cases, so rewrite it
```suggestion
    if (todoOnEdit) {
      setTodoOnEdit(null);
      if (todoOnEdit.title !== titleQuery) {
        rtodosUpdater(titleChanger(todos, todoOnEdit, titleQuery));
      }
    }
```